hubtools/src/lib.rs

```diff
@@ -796,4 +808,43 @@ impl RawHubrisArchive {
     pub fn replace(&mut self, data: Vec<u8>) {
         self.image.replace(data);
     }
+
+    pub fn verify(
+        &mut self,
+        src_cmpa: PathBuf,
+        src_cfpa: PathBuf,
```

Comment:
API nit - I think these should probably be either `&[u8]` or even `[u8; 512]`. In the wicket case we will have fetched these via MGS and have them in memory already, right? If we make them `[u8; 512]` we can also ditch the new "bad length" error variants, since we force the callers to size them correctly.